tests/integration-service/namespace-backed-environments.go

```diff
@@ -240,10 +241,17 @@ var _ = framework.IntegrationServiceSuiteDescribe("Namespace-backed Environment
 
 		When("Integration Test PipelineRun completes successfully", func() {
 			It("should lead to Snapshot CR being marked as passed", FlakeAttempts(3), func() {
-				Eventually(func() bool {
+				Eventually(func() error {
 					snapshot, err = f.AsKubeAdmin.IntegrationController.GetSnapshot("", pipelineRun.Name, "", testNamespace)
-					return err == nil && f.AsKubeAdmin.CommonController.HaveTestsSucceeded(snapshot)
-				}, time.Minute*5, time.Second*5).Should(BeTrue(), fmt.Sprintf("Timed out waiting for Snapshot to be marked as succeeded %s/%s", snapshot.GetNamespace(), snapshot.GetName()))
+					if err != nil {
+						GinkgoWriter.Println("snapshot has not been found yet")
+						return err
+					}
+					if !f.AsKubeAdmin.CommonController.HaveTestsSucceeded(snapshot) {
+						return fmt.Errorf("tests haven't succeeded for snapshot %s/%s. snapshot status: %+v", snapshot.GetNamespace(), snapshot.GetName(), snapshot.Status)
+					}
+					return nil
+				}, time.Minute*5, time.Second*5).Should(Succeed(), fmt.Sprintf("timed out waiting for the Snapshot for the Integration Test PipelineRun %s/%s to be marked as succeeded", testNamespace, pipelineRun.Name))
```

Comment:
```suggestion
				}, time.Minute*5, time.Second*5).Should(Succeed(), fmt.Sprintf("timed out waiting for the Snapshot for the Build PipelineRun %s/%s to be marked as succeeded", testNamespace, pipelineRun.Name))
```
since the `pipelineRun` variable contains Build PLR that triggered the creation of this Snapshot.

Reply:
@rsoaresd In your latest commit, can you please replace "Build Test PipelineRun" with "Build PipelineRun"?